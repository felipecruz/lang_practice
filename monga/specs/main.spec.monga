void teste (int x, int y) {
    a = new Space();
    x = x + 1;
    return x;
}

int quickort (int array[], int length)
{
    int middle = length / 2;
    int pivot = array[middle];

    left = partition (array, pivot, "lesser");
    right = partition (array, pivot, "greater");

    return join (quickort (left), {pivot}, quickort (right));
}

int main (int argc, char *argv) {
    int array[] = {0, 1, 2};

    int a = 10;
    int b = 100;
    int d = 0xF1;
    int x = 1.2E+09;
    int d = .9;
    int y = 1.2;

    a = 1;
    a + a;
    a = 1 + a;

    b = (c + b);
    b = (a * b) + ((c + b) * f);

    a = (1 == 2) ? 2 : 3;

    while (1 == 2 || b > x - 2) {
        printf ("%");
    }
}

