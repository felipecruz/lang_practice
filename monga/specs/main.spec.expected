
K_TYPE_VOID: void 
TK_ID: teste 
TK_TYPE_INT: int 
TK_ID: x 
TK_TYPE_INT: int 
TK_ID: y 
    
TK_ID: a  
K_NEW: new 
TK_ID: Space
    
TK_ID: x  
TK_ID: x  
TK_NUMBER: 1
    
K_RETURN: return 
TK_ID: x



TK_TYPE_INT: int 
TK_ID: quickort 
TK_TYPE_INT: int 
TK_ID: array 
TK_TYPE_INT: int 
TK_ID: length

    
TK_TYPE_INT: int 
TK_ID: middle  
TK_ID: length  
TK_NUMBER: 2
    
TK_TYPE_INT: int 
TK_ID: pivot  
TK_ID: array
TK_ID: middle

    
TK_ID: left  
TK_ID: partition 
TK_ID: array 
TK_ID: pivot 
TK_STRING: "lesser"
    
TK_ID: right  
TK_ID: partition 
TK_ID: array 
TK_ID: pivot 
TK_STRING: "greater"

    
K_RETURN: return 
TK_ID: join 
TK_ID: quickort 
TK_ID: left 
TK_ID: pivot 
TK_ID: quickort 
TK_ID: right



TK_TYPE_INT: int 
TK_ID: main 
TK_TYPE_INT: int 
TK_ID: argc 
K_TYPE_CHAR: char 
TK_ID: argv 
    
TK_TYPE_INT: int 
TK_ID: array  
TK_NUMBER: 0 
TK_NUMBER: 1 
TK_NUMBER: 2

    
TK_TYPE_INT: int 
TK_ID: a  
TK_NUMBER: 10
    
TK_TYPE_INT: int 
TK_ID: b  
TK_NUMBER: 100
    
TK_TYPE_INT: int 
TK_ID: d  
TK_HEXA: 0xF1
    
TK_TYPE_INT: int 
TK_ID: x  
TK_FLOAT: 1200000000.000000
    
TK_TYPE_INT: int 
TK_ID: d  
TK_FLOAT: 0.900000
    
TK_TYPE_INT: int 
TK_ID: y  
TK_FLOAT: 1.200000

    
TK_ID: a  
TK_NUMBER: 1
    
TK_ID: a  
TK_ID: a
    
TK_ID: a  
TK_NUMBER: 1  
TK_ID: a

    
TK_ID: b  
TK_ID: c  
TK_ID: b
    
TK_ID: b  
TK_ID: a  
TK_ID: b  
TK_ID: c  
TK_ID: b  
TK_ID: f

    
TK_ID: a  
TK_NUMBER: 1  
TK_NUMBER: 2  
TK_NUMBER: 2  
TK_NUMBER: 3

    
K_WHILE: while 
TK_NUMBER: 1  
TK_NUMBER: 2  
TK_ID: b  
TK_ID: x  
TK_NUMBER: 2 
        
TK_ID: printf 
TK_STRING: "%"
    


